api/server/router/local/image.go

```diff
@@ -284,6 +294,18 @@ func (s *router) postBuild(ctx context.Context, w http.ResponseWriter, r *http.R
 	w.Header().Set("Content-Type", "application/json")
 
 	version := httputils.VersionFromContext(ctx)
+	output := ioutils.NewWriteFlusher(w)
+	sf := streamformatter.NewJSONStreamFormatter()
+	errf := func(err error) error {
+		// Do not write the error in the http output if it's still empty.
+		// This prevents from writing a 200(OK) when there is an interal error.
+		if !output.Flushed() {
+			return err
+		}
+		w.Write(sf.FormatError(errors.New(utils.GetErrorMessage(err))))
+		return nil
```

Comment:
@LK4D4 think of it this way: the logic in `errf` is always correct: only send the error code back from the handler if the HTTP header wasn't already written (the output.Flushed() condition). If it was already sent, then the error doesn't matter at all that's why it's nil. In that case we have to write the error in the body. So technically the json encode error should have the same behavior as the previous code.